build-common/src/org/jetbrains/kotlin/incremental/IncrementalJvmCache.kt

```diff
@@ -646,83 +647,86 @@ class KotlinClassInfo constructor(
-            if (access and Opcodes.ACC_PRIVATE == Opcodes.ACC_PRIVATE) return null
+private class ConstantsClassVisitor : ClassVisitor(Opcodes.API_VERSION) {
+    private val result = LinkedHashMap<String, Any>()
 
-            val staticFinal = Opcodes.ACC_STATIC or Opcodes.ACC_FINAL
-            if (value != null && access and staticFinal == staticFinal) {
-                result[name] = value
-            }
-            return null
+    override fun visitField(access: Int, name: String, desc: String, signature: String?, value: Any?): FieldVisitor? {
+        if (access and Opcodes.ACC_PRIVATE == Opcodes.ACC_PRIVATE) return null
+
+        val staticFinal = Opcodes.ACC_STATIC or Opcodes.ACC_FINAL
+        if (value != null && access and staticFinal == staticFinal) {
+            result[name] = value
         }
-    }, ClassReader.SKIP_CODE or ClassReader.SKIP_DEBUG or ClassReader.SKIP_FRAMES)
+        return null
+    }
 
-    return result
+    fun getResult() = result
 }
 
-private fun getInlineFunctionsMap(header: KotlinClassHeader, bytes: ByteArray): LinkedHashMap<String, Long> {
-    val inlineFunctions = inlineFunctionsJvmNames(header)
-    if (inlineFunctions.isEmpty()) return LinkedHashMap()
-
-    val result = LinkedHashMap<String, Long>()
-    var dummyVersion: Int = -1
-    ClassReader(bytes).accept(object : ClassVisitor(Opcodes.API_VERSION) {
-
-        override fun visit(
-            version: Int,
-            access: Int,
-            name: String?,
-            signature: String?,
-            superName: String?,
-            interfaces: Array<out String>?
-        ) {
-            super.visit(version, access, name, signature, superName, interfaces)
-            dummyVersion = version
-        }
-
-        override fun visitMethod(
-            access: Int,
-            name: String,
-            desc: String,
-            signature: String?,
-            exceptions: Array<out String>?
-        ): MethodVisitor? {
-            if (access and Opcodes.ACC_PRIVATE == Opcodes.ACC_PRIVATE) return null
-
-            val dummyClassWriter = ClassWriter(0)
-            dummyClassWriter.visit(dummyVersion, 0, "dummy", null, AsmTypes.OBJECT_TYPE.internalName, null)
-
-            return object : MethodVisitor(Opcodes.API_VERSION, dummyClassWriter.visitMethod(0, name, desc, null, exceptions)) {
-                override fun visitEnd() {
-                    val jvmName = name + desc
-                    if (jvmName !in inlineFunctions) return
-
-                    val dummyBytes = dummyClassWriter.toByteArray()!!
-
-                    val hash = dummyBytes.md5()
-                    result[jvmName] = hash
-                }
+private class InlineFunctionsClassVisitor(
+    private val inlineFunctionNames: Set<String>,
+    cv: ClassVisitor // Note: cv must not override the visitMethod (it will not be called with the current implementation below)
```

Comment:
Use `ConstantsClassVisitor` as param type to be more specific.

Reply:
Done